web/src/app/services/data-store/data-store.service.ts

```diff
@@ -188,7 +187,13 @@ export class DataStoreService {
   ): Promise<void> {
     const {id: surveyId, jobs} = survey;
 
-    jobIdsToDelete?.forEach(jobId => this.deleteJob(surveyId, jobId));
+    if (jobIdsToDelete) {
+      await this.db.firestore.runTransaction(async transaction => {
+        for (const jobId of jobIdsToDelete) {
```

Comment:
The entire operation `updateSurvey` should be done in the same transaction to ensure atomicity and isolation, not just deleting jobs.